### PR TITLE
feat(npu): add Qwen3.5-35B-A3B training scripts and NPU infrastructure

### PR DESCRIPTION
**Pull request by hbamboo**
# ⭐ Feature
## Qwen3.5-35B-A3B training scripts
- Add 16xNPU fully-async training script with GRPO support
  - TP=4, EP=8, `--num-iters-per-train-update 32`, max-staleness=2
  - Actor (8 GPUs) + Rollout (8 GPUs) via independent placement groups
- Add 16xNPU colocate training script
  - Actor and Rollout time-share 16 GPUs via `--colocate`
- Add 8xNPU colocate training script (single-node)
## Qwen3.5-9B script tuning
- Colocate: add `--max-actor-ckpt-to-keep 1` to limit disk usage
- Async: tune SGLang params (mem-fraction 0.7→0.8, cuda-graph-bs refined, add `--sglang-max-running-requests 256`)
---
# 🔩 NPU Infrastructure
## Docker
- Pin TransferQueue to commit `dcc78f0` for build reproducibility
- Add `protobuf==6.33.6` pin to avoid compatibility issues
- Apply new `sglang-npu.patch` during SGLang build
- Fix MindSpeed-Bridge clone indentation
## NPU Patches
- **sglang-npu.patch** (new): NPU-specific fixes for SGLang inference engine
- **megatron.patch**: Add gate slicing fix for `num_query_groups < world_size` in attention module
- **mindspeed-bridge.patch**: Refactor and expand NPU bridge compatibility
---
# 🐛 Bug Fix
- Set `partition_dim=0` for NPU in `weight_update/common.py` `linear_fc1.weight` all-gather path to work around Megatron grouped MoE bug
---
# 📝 Documentation
- Update `npu-training.md`: add Qwen3.5-35B-A3B to model support table, expand device list to 16 cards, update roadmap

- 300step results compare with H800:
 - colocate:
 raw-reward-qwen35-35B
- async:

raw-reward-qwen35-35B-async

### CODE REVIEW

File: relax/backends/megatron/weight_update/common.py [hunk 1]
```diff
@@ -110,6 +111,8 @@ def all_gather_param(args, name: str, param: torch.nn.Parameter) -> torch.Tensor
     if "linear_fc1.weight" in name and "vision_model" not in name:
         param_partitions = [p.chunk(2, dim=0) for p in param_partitions]
         param_partitions = [p[0] for p in param_partitions] + [p[1] for p in param_partitions]
+        if is_npu_available:
```
Comment (NINGBENZHE) [inline relax/backends/megatron/weight_update/common.py:115]: 这块权重转换逻辑，npu为什么不一样？

File: scripts/training/text/run-qwen35-35B-A3B-16xnpu-async.sh [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+# Copyright (c) 2026 Relax Authors. All Rights Reserved.
+#
+# Qwen3.5-35B-A3B 16xGPU (2-node) fully async training script for DAPO math dataset.
+#
+# Usage:
+#   bash scripts/training/text/run-qwen35-35B-A3B-16xgpu-async.sh
+set -ex
+set -o pipefail
+export HCCL_SOCKET_IFNAME=enp23s0f3
```
Comment (NINGBENZHE) [inline scripts/training/text/run-qwen35-35B-A3B-16xnpu-async.sh]: 这种网卡的参数配置不要写死，改成类似
export HCCL_SOCKET_IFNAME="${HCCL_SOCKET_IFNAME:-enp23s0f3}"

File: scripts/training/text/run-qwen35-35B-A3B-16xnpu-colocate.sh [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+# Usage:
+#   NUM_GPUS=4 bash scripts/training/text/run-qwen3-4B-4xgpu-async.sh
+
+set -ex
+set -o pipefail
+unset http_proxy https_proxy HTTP_PROXY HTTPS_PROXY
+
+ulimit -n 65535
+
+export HCCL_SOCKET_IFNAME=enp23s0f3
```
Comment (NINGBENZHE) [inline scripts/training/text/run-qwen35-35B-A3B-16xnpu-colocate.sh]: 同上

File: relax/backends/megatron/weight_update/common.py (hunk 1 above)
Comment (hbamboo) [reply relax/backends/megatron/weight_update/common.py:115]: partition_dim gpu是0，npu是1，导致shape不匹配，先修改规避

**Review by NINGBENZHE — APPROVED** [review]
/lgtm
